hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/ExecutionContext.java

```diff
@@ -129,13 +137,19 @@ public ExecutionContext initialize(ExecutionPlan plan) {
         snapshotContext = new SnapshotContext(nodeEngine.getLogger(SnapshotContext.class), jobNameAndExecutionId(),
                 plan.lastSnapshotId(), jobConfig.getProcessingGuarantee());
 
+        serializationService = new JetSerializationService(
```

Comment:
Maybe we can avoid this if there are no serializers configured.

Reply:
We need to call `destroy()` at job completion, that would require another `if` in `completeExecution`.